starfyre/compiler.py

```diff
@@ -153,6 +163,7 @@ def fx_{root_name}():
     return component
```

Comment:
why not just do this
```
return render_root(component)

rendered_{root_name} = fx_{root_name}()
```

Also, I don't think that I understand it completely. I will have a proper read tomorrow morning 😄

Reply:
It is okay to do it this way:
```python
return component
```
but we will have to resort to the former way of rendering the pages. Like this:
```python
result = str(render_root(component)) if not route_name == 'app' else component # we don't want this.
``` 


So now what I have done is:
- ensure that each file parsed is treated as a component(i.e a `Component` child)
- ensure we have an output of `render_root` for each of our pages that are being transpiled.

Why?
It will help the parser be able to resolve & recognise our imports. I noticed that the parser could actually read the contents of the file while I was debugging but it could not resolve these imports(`store` and `parent`) as children of the `Component` class, which they are. This was what was causing the issue.

So that's why I decided that, we let each page being transpiled to be a normal `Component` but also keep the output of `render_root(page_being_transpiled)`. This output will be used to write to the individual pages.

Reply:
>  I noticed that the parser could actually read the contents of the file while I was debugging but it could not resolve these imports(store and parent) as children of the Component class, which they are

What was the issue here?